### PR TITLE
Document support for asymmetric ID Token Key Management Encryption Al…

### PR DESCRIPTION
**Pull request by chris468**
…gorithms

- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/systemlink-operations-handbook/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Documents configuration required to use asymmetric encryption algorithms for ID token key management encryption.

### Why should this Pull Request be merged?

Allows customers to use additional encryption options.

### What testing has been done?

- Verified both of the added options work
- Built documentation and verified formatting

### CODE REVIEW

**Review by spanglerco — CHANGES_REQUESTED** [review]

File: handbook/openid-connect/openid-connect.md [hunk 1]
```diff
@@ -178,6 +197,8 @@ SystemLink supports the following algorithms for ID token signing, ID token key
 
 ### ID Token Key Management Encryption Algorithm
 
+Algorithms which do not require a private key:
```
Comment (jjaco2) [inline handbook/openid-connect/openid-connect.md:200]: Here and below: change 'which' to 'that'

File: handbook/openid-connect/openid-connect.md [hunk 2]
```diff
@@ -338,3 +367,22 @@ To resolve this issue:
 3. Modify the number at the end to a number larger than X, where X is the required size of the cache entry specified in the error log.
 
 4. Restart the NI Web Server from the `Control` tab of the NI Web Server Configuration application.
+
+### ID Token Management Encryption
+
+If the provider is configured to use an asymmmetric ID token management encryption algorithm and private keys are missing or incorrect, or if the provider is configured to use an unsupported ID token management encryption algorithm, the user will be redirected back to the SystemLink login page after logging in to the provider.
```
Comment (jjaco2) [inline handbook/openid-connect/openid-connect.md:373]: Lots of good info here but kind of hard to parse. Maybe something like:

'The following situations can lead to an error and redirect the user back to the SystemLink login page after logging in to the provider:

- The provider is configured to use an asymmetric...

- The provider is configured to use an unsupported ID...'

File: handbook/openid-connect/openid-connect.md [hunk 3]
```diff
@@ -338,3 +367,22 @@ To resolve this issue:
...
+
+!!! note "Example error logs"
+    When this happens, the NI Web Server error logs will contain entries containing the following:
```
Comment (jjaco2) [inline handbook/openid-connect/openid-connect.md:376]: > When this happens,

Opportunity to clarify whether it's the act of redirecting the user that logs the error (which is what it sounds like now), or something else

File: handbook/openid-connect/openid-connect.md [hunk 4]
```diff
@@ -338,3 +367,22 @@ To resolve this issue:
...
+
+    ```text
+    oidc_proto_parse_idtoken: oidc_jwt_parse failed
+    ```
+
+    The log entry will also contain more information about the problem.
+
+To resolve this issue:
+
+1. Confirm that the provider is configured to use supported encryption and signing algorithms. See [**Supported Signing and Encryption Algorithms**](#supported-signing-and-encryption-algorithms).
```
Comment (jjaco2) [inline handbook/openid-connect/openid-connect.md:386]: Would 'uses' or 'is using' do the same work here? 

I get the feeling we want to drive home that this is all configurable. In that case, would saying 'In [the configuration utility or something], confirm the provider is using supported encryption...' work to give them more direct resolution guidance?

File: handbook/openid-connect/openid-connect.md [hunk 5]
```diff
@@ -188,6 +209,14 @@ SystemLink supports the following algorithms for ID token signing, ID token key
 
 - AES-256 Key Wrap
 
+Algorithms which require a private key:
+
+- RSAES OAEP
+
+- ECDH-ES
+
+Refer to [**SystemLink Login Configuration**](#systemlink-login-configuration) for information on configuring the private key.
```
Comment (cameronwaterman) [inline handbook/openid-connect/openid-connect.md:218]: I assume this link works better when running via mkdocs? Following that link in the github md previewer produces some weird behavior. Same for other links.

Comment (chris468) [reply handbook/openid-connect/openid-connect.md:218]: For me they are working both in mkdocs and when you're looking at the rendered file in the [repo](https://github.com/ni/systemlink-operations-handbook/blob/configure-asymmetric-id-token-encryption-key/handbook/openid-connect/openid-connect.md). It didn't work in the "rich diff" view in the PR.

**Review by cameronwaterman — APPROVED** [review]

File: handbook/openid-connect/openid-connect.md (hunk 4 above)
Comment (chris468) [reply handbook/openid-connect/openid-connect.md]: I referred to the provider's documentation b/c the details of where to check the configuration is going to depend on the provider. Let me know what you think.

**Review by spanglerco — APPROVED** [review]

**Review by prestwick — APPROVED** [review]
